### ISSUE TITLE
workflow and functionality for publication of additional data 

### ISSUE THREAD

**Issue by IngoBethke** [NorwegianClimateCentre/noresm2cmor#99]
Our current practice concerning adding new fields for an experiments is we define a new version for cmor-isation, append the new files to the .sha256sum files and regenerate the .map files. Given that the .map files contain both published and not yet published file entries it is not clear how to use the .map files for publishing only the new files. 

In NorCPM1 publication workflow I have therefore made some small changes to facilitate publication of additional data:

Instead of appending checksums to existing .sha256sum files, a new .sha256sum file is created for every version and the version id is added to the end of the .sha256sum file name e.g. .r1i1p1f1.sha256sum_v20191005 (see https://github.com/NorwegianClimateCentre/noresm2cmor/blob/master/scripts/create_CMIP6_sha256sum_NorCPM1.sh).    

If the environmental variable VERSION is set to YYYYMMDD, create_CMIP6_maps.sh will search for .sha256sum_vYYYYMMDD files and create map files with extension .map_vYYYYMMDD that can be used for publishing the additional data. 

@YanchunHe , @monsieuralok : How do you currently handle publication of additional data? Could it be an idea to do something similar to the above?

**Comment by YanchunHe** [comment]
Thanks Ingo for put this issue on table.

Indeed, so far I append all the new cmrozed files with different data version to the previous .sha256sum. I also like the idea to put different dataversion to different .sha256sum.$VERSION. That also make it eaiser for me to update the sha256sum.

Now I have a even different situation. Some have different simulations but for the same 'experiment', and I would like to keep the new cmorized files with the same data version because the are same number of cmorized fields, and I want to keep as less as data version as possible.

We can think if we can put as many as seperate sha256sum (thus map files) as we can in a simple way. Can you indentify easily which .sha256sum files have published and which not?

**Comment by monsieuralok** [comment]
Hi,
        I have created a script when I was having this issue with PAMIP and new variables were added. I used following line in a script
grep -Fxvf ../old_map_file new_map_file >new_publishdata_map

and created new map file for myself. Also, it could be better to have only sha256sum based on new inserted data.

**Comment by YanchunHe** [comment]
I will use different  .sha256sum_$VERSION files for different versions.
But I may use the same data version for several cmorization jobs. Therefore, in this case, new sha256sum will append to the existing sha256sum_$Version file. And Alok need create a seperate map to find which are new files to be published.
commit: 50ac095

**Comment by IngoBethke** [comment]
I agree that it makes sense to reserve versioning for correcting data and use the same version if merely diagnostics are added. 

Still, I'd think it is cleaner if the checksum and map files are left unchanged once created. Changing them will it make more difficult to backtrack what happened at a later stage. 

My suggestion would to add a letter after the version in the checksum and map file names if multiple batches exist. Maybe the the letter "a" could be skipped e.g. _v20191005 , _v20191005b, _v20191005c. I think this should be relatively straight forward to implement. What do you think?

**Comment by YanchunHe** [comment]
Thanks Ingo!

A new script (`create_CMIP6_links_sha256sum.sh`) creates the links and sha256sum at the same time, and only deal with each single data version folder under .cmorout. A separate .sha256sum_version file will be created and only contains the  unpublished data. 
commit: 3df554915f35f2441e66d8509364ce686aa66ffe

During cmorization, I also changed slightly the format of the data version. The same data version is reserved for the same experiment but with the same var.nml (and other settings). However, the the folder will also be add a suffix, e.g, _v20191005b if a version _v20191005 has been published.

Refer to Issue #94

**Comment by YanchunHe** [comment]
If you don't find there needs further improve on this, I will close this issue.
